### PR TITLE
Publish Releases

### PR DESCRIPTION
**Pull request by scotwells**
## Summary
This PR uses [Goreleaser](https://goreleaser.com) to publish binaries, linux packages, and homebrew formula manifests for the Datum Cloud CLI. 

- Release artifacts are published to the GitHub Release
  - Software Binaries
  - SBOM
  - Linux packages
    - apk
    - deb
    - rpm
    - termux.deb
    - archlinux
- Homebrew forumula manifest is pushed to [datum-cloud/homebrew-tap](https://github.com/datum-cloud/homebrew-tap)

## Installing via Homebrew

This example was based on an example release that was published.

```shell
$ brew tap datum-cloud/tap
==> Tapping datum-cloud/tap
Cloning into '/opt/homebrew/Library/Taps/datum-cloud/homebrew-tap'...
...

$ brew install datum-cloud/tap/cli
==> Fetching datum-cloud/tap/cli
==> Downloading https://github.com/datum-cloud/datumctl/releases/download/v0.0.1-rc9/datumctl_Darwin_arm64.tar.gz
...

$ datumctl --help
A CLI for Datum Cloud

Usage:
  datumctl [command]

Available Commands:
  auth        Authenticate with Datum Cloud
  completion  Generate the autocompletion script for the specified shell
  help        Help about any command

Flags:
  -h, --help   help for datumctl

Use "datumctl [command] --help" for more information about a command.
```

### CODE REVIEW

File: .goreleaser.yaml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+# https://goreleaser.com
+#
+# yaml-language-server: $schema=https://goreleaser.com/static/schema.json
+# vim: set ts=2 sw=2 tw=0 fo=cnqoj
+version: 2
+
+before:
+  hooks:
+    - go mod tidy
+    - go generate ./...
```
Comment (joshlreese) [inline .goreleaser.yaml:11]: It seems odd to have `go generate` called at build time. I suspect this was from a template that assumes generated content is not checked into the repo?

File: .goreleaser.yaml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      - "^test:"
+      - "^chore:"
+
+release: {}
+
+sboms:
+  - artifacts: archive
+
+brews:
+- name: cli
```
Comment (joshlreese) [inline .goreleaser.yaml:66]: Should we have this match the tool's name, `datumctl`?

File: .goreleaser.yaml (hunk 1 above)
Comment (scotwells) [reply .goreleaser.yaml:11]: Correct, this was just included from the example template they had.

File: .goreleaser.yaml (hunk 2 above)
Comment (scotwells) [reply .goreleaser.yaml:66]: I used that originally but the brew command to install the tool would be `brew install datum-cloud/tap/datumctl` and I didn't like the stuttering. Using `cli` as the name here makes the install command `brew install datum-cloud/tap/cli`.

Comment (joshlreese) [reply .goreleaser.yaml:66]: I'm not super familiar with brew, but how does one get to the point of `brew install datumctl`? Does this naming choice introduce any barriers there?

I noticed that Pulumi uses `pulumi/tap/pulumi`, but GitHub's CLI is `brew install gh`.

Comment (scotwells) [reply .goreleaser.yaml:66]: The format `pulumi/tap/pulumi` automatically discovers the brew formula to install the package. The format looks for a formula named `pulumi` in the repo `pulumi/hombrew-tap` (e.g. [pulumi tap formula](https://github.com/pulumi/homebrew-tap/blob/main/Formula/pulumi.rb)). To do `brew install gh` you have to make a PR with a formula to the hombrew core repo (e.g. [GitHub formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/g/gh.rb)).

Comment (joshlreese) [reply .goreleaser.yaml:66]: Looks like if you use `brew tap <user>/<repo>` that naming conflicts would require you install with a fully qualified name, per https://docs.brew.sh/Taps#formula-with-duplicate-names. I'm not sure what other brew packages we may have in the future, but having a name like `cli` may make adding the tap ahead of time pointless.

I'm not sure how often users do that though, so until we're closer to a v1 here I think I'm ok with `cli` for now.

**Review by joshlreese — APPROVED** [review]

**Review by joshlreese — APPROVED** [review]

**Pull request merged**